### PR TITLE
docs: document bot verification

### PR DESCRIPTION
**Pull request by e-moran**
Adds documentation for Arcjet's bot verification feature.

It only updates examples that are configured in an "allow" mode, because that is how we recommend using the feature.

### CODE REVIEW

File: src/snippets/bot-protection/quick-start/bun/Step3.js [hunk 1]
```diff
@@ -27,6 +27,10 @@ export default {
       return new Response("Forbidden", { status: 403 });
     }
 
+    if (decision.reason.isBot() && decision.reason.isSpoofed()) {
```
Comment (davidmytton) [inline src/snippets/bot-protection/quick-start/bun/Step3.js:30]:
```suggestion
    if (decision.reason.isBot() || decision.reason.isSpoofed()) {
```

Shouldn't all these be OR? That's how you used it in [the SDK repo examples](https://github.com/arcjet/arcjet-js/pull/2375/files#diff-d87647aacb2ee3bb9ef0c144004d6db6d13443b0aa0ff79fdbb83e3474a263a3R37).

File: src/snippets/bot-protection/reference/nextjs/AllowingBotsApp.js [hunk 2]
```diff
@@ -22,16 +22,28 @@ const aj = arcjet({
...
-      {
-        error: "You are a bot!",
-        // Useful for debugging, but don't return these to the client in
-        // production
-        denied: decision.reason.denied,
-      },
-      { status: 403 },
-    );
+  if (decision.reason.isBot()) {
+    if (decision.isDenied()) {
```
Comment (davidmytton) [inline src/snippets/bot-protection/reference/nextjs/AllowingBotsApp.js:26]: Why have these separate when it could be OR so it's fewer lines of code?

File: src/content/docs/bot-protection/reference.mdx [hunk 3]
```diff
@@ -329,6 +331,32 @@ their "advertising quality" bot.
   <SvelteKitFiltering slot="sveltekit" />
 </SlotByFramework>
 
+## Bot verification
+
+We support verifying the authenticity of requests from identifiable bots by
```
Comment (davidmytton) [inline src/content/docs/bot-protection/reference.mdx:336]: Can you rephrase this slightly so it's clear we're doing this behind the scenes for you and you don't need to do the lookups yourself?

**Review by davidmytton — APPROVED** [review]

**Review by davidmytton — CHANGES_REQUESTED** [review]
Added some comments to clarify things.

File: src/snippets/bot-protection/quick-start/bun/Step3.js (hunk 1 above)
Comment (blaine-arcjet) [reply src/snippets/bot-protection/quick-start/bun/Step3.js:30]: This needs to be `&&` because the type needs to narrow on `isBot()`. The example you linked has `isBot()` as a separate `if` statement wrapping the section that checks `isSpoofed()` because it is "denied or spoofed"

File: src/content/docs/bot-protection/reference.mdx [hunk 4]
```diff
@@ -283,8 +287,6 @@ properties.
 </SlotByFramework>
 
 ## Error handling
-
-Arcjet is designed to fail open so that a service issue or misconfiguration does
```
Comment (blaine-arcjet) [inline src/content/docs/bot-protection/reference.mdx:287]: Was this accidentally removed?

File: src/content/docs/bot-protection/reference.mdx [hunk 5]
```diff
@@ -329,6 +331,32 @@ their "advertising quality" bot.
...
+DNS query. To check the verification status, use the `.isSpoofed()` or `.isVerified()`
+methods on the reason object. If verification was performed, one of these will be true
+and the other false; both will be false if verification failed or is unsupported
+for the identified bot. 
+
+:::note
+It's recommended to check verification status only when using an allow configuration.
+For deny configurations, verification is skipped as you'd want to block listed bots
+regardless of legitimacy.
+:::
```
[the remark this answers is not in the document]
Comment (davidmytton) [reply src/content/docs/bot-protection/reference.mdx:347]: I'm not sure it's that's useful and is an extra step we can skip. It might be interesting to show "requests from bogus/spoofed crawlers" in analytics, but we could calculate that out of band.

[the remark this answers is not in the document]
Comment (e-moran) [reply src/content/docs/bot-protection/reference.mdx:350]: Yeah I agree, I think its best to return a decision to the user as soon as possible; if the bot's ip isn't cached it would mean we're performing a relatively expensive lookup for no apparent value.

File: src/content/docs/bot-protection/reference.mdx (hunk 3 above)
Comment (e-moran) [reply src/content/docs/bot-protection/reference.mdx]: I've added an extra sentence to explain it, let me know what you think!

Comment (davidmytton) [reply src/content/docs/bot-protection/reference.mdx]: Looks good, thanks.

File: src/content/docs/bot-protection/reference.mdx (hunk 5 above)
[the remark this answers is not in the document]
Comment (blaine-arcjet) [reply src/content/docs/bot-protection/reference.mdx:350]: The problem I see is that "almost behaving bots" will avoid detection by changing their user agent strings and then we won't do IP lookup. We've seen this with AllenAI in that they released "Dolma" to change the user agent string and avoid previous detections—they could have deployed to different IP ranges but I'm guessing they used the same ones.

[the remark this answers is not in the document]
Comment (e-moran) [reply src/content/docs/bot-protection/reference.mdx:350]: I think it is a fair point, but I think that's a future expansion of the detect bots feature rather than being related to the bot verification product. We can use the same tools that we've built out in this project to perform this kind of detection though. We can open an issue in the main repo to track this.

File: src/snippets/bot-protection/reference/nextjs/AllowingBotsApp.js (hunk 2 above)
Comment (e-moran) [reply src/snippets/bot-protection/reference/nextjs/AllowingBotsApp.js:26]: Changed it to that, I missed that the contents of the blocks were the same 🤦

File: src/content/docs/bot-protection/reference.mdx [hunk 6]
```diff
@@ -329,6 +333,33 @@ their "advertising quality" bot.
   <SvelteKitFiltering slot="sveltekit" />
 </SlotByFramework>
 
+## Bot verification
```
Comment (davidmytton) [inline src/content/docs/bot-protection/reference.mdx:336]: This section needs to be added to the table of contents.

File: src/content/docs/bot-protection/reference.mdx [hunk 7]
```diff
@@ -168,6 +168,10 @@ Most applications want to block almost all bots. However, it is common to allow
 some bots to access your system, such as bots for search indexing or API
 access from the command line.
 
+When allowing specific bots we recommend that you also check the verification
```
Comment (davidmytton) [inline src/content/docs/bot-protection/reference.mdx:171]:
```suggestion
When allowing specific bots we recommend that you also [check the verification](#bot-verification)
```

File: src/content/docs/bot-protection/reference.mdx [hunk 8]
```diff
@@ -329,6 +333,33 @@ their "advertising quality" bot.
   <SvelteKitFiltering slot="sveltekit" />
 </SlotByFramework>
 
+## Bot verification
+
+We support verifying the authenticity of requests from some bots by
+comparing the request IP to published bot IP ranges or performing a reverse
+DNS query. Verificaiton is performed for supported bots using the appropriate method
+automatically after the bot is identified by Arcjet. To check the verification status,
```
Comment (davidmytton) [inline src/content/docs/bot-protection/reference.mdx:341]:
```suggestion
automatically after the bot is identified by Arcjet. 

To check the verification status,
```

(make this a new paragraph)

**Review by davidmytton — CHANGES_REQUESTED** [review]

File: src/content/docs/bot-protection/reference.mdx [hunk 9]
```diff
@@ -329,6 +335,35 @@ their "advertising quality" bot.
   <SvelteKitFiltering slot="sveltekit" />
 </SlotByFramework>
 
+## Bot verification
+
+We support verifying the authenticity of requests from some bots by
+comparing the request IP to published bot IP ranges or performing a reverse
+DNS query. Verificaiton is performed for supported bots using the appropriate method
```
Comment (davidmytton) [inline src/content/docs/bot-protection/reference.mdx:342]:
```suggestion
DNS query. Verification is performed for supported bots using the appropriate method
```

**Review by davidmytton — APPROVED** [review]

File: src/content/docs/bot-protection/reference.mdx (hunk 5 above)
[the remark this answers is not in the document]
Comment (e-moran) [reply src/content/docs/bot-protection/reference.mdx:354]: https://github.com/arcjet/arcjet/issues/3666
